azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -36,14 +36,15 @@ class AbstractIoTHubClient(object):
     This class needs to be extended for specific clients.
     """
 
-    def __init__(self, iothub_pipeline):
+    def __init__(self, iothub_pipeline, upload_pipeline):
         """Initializer for a generic client.
 
         :param iothub_pipeline: The pipeline used to connect to the IoTHub endpoint.
         :type iothub_pipeline: :class:`azure.iot.device.iothub.pipeline.IoTHubPipeline`
         """
         self._iothub_pipeline = iothub_pipeline
-        self._edge_pipeline = None
+        self._upload_pipeline = upload_pipeline
```

Comment:
>self._upload_pipeline = upload_pipeline [](start = 8, length = 39)

Not sure Modules support Upload right now. Maybe this has changed since I last heard? It was a fairly arbitrary restriction at the time.